src/System.CodeDom/src/System/CodeDom/Compiler/CompilerInfo.cs

```diff
@@ -103,7 +103,7 @@ internal CompilerInfo(CompilerParameters compilerParams, string codeDomProviderT
         public override bool Equals(object o)
         {
             CompilerInfo other = o as CompilerInfo;
-            if (o == null)
+            if (other == null)
```

Comment:
The fact that all of our tests were passing before this change suggests that we're missing some test coverage.  To go along with your fixes, can you add tests that would have failed before the changes?